### PR TITLE
Syndics honor MoM acl

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

This is an alternate approach to #63257

### What issues does this PR fix or reference?
Fixes: 

#62618
#62577

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/utils/test_minions.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def test_validate_tgt_should_return_false_when_no_valid_minions_have_been_found(
         "salt.utils.minions.CkMinions.check_minions", autospec=True, return_value={}
     ):
         result = ckminions.validate_tgt("fnord", "fnord", "fnord", minions=[])
-        assert result is False
```
Comment (waynew) [inline tests/pytests/unit/utils/test_minions.py:68]: FWIW this test should either be renamed or completely removed - this was part of the CVE check to ensure that we weren't validating minions that weren't valid.

**Review by waynew — CHANGES_REQUESTED** [review]
Only one problem that I see, and that's in the test suite.

I'm going to check this out and bang on it a bit more. I'm also seeing some failures here: https://github.com/saltstack/salt/actions/runs/3864153908/jobs/6586792092 that are concerning

File: tests/pytests/unit/utils/test_minions.py (hunk 1 above)
Comment (dwoz) [reply tests/pytests/unit/utils/test_minions.py:68]: Agreed, I've renamed the test and also added a doc string to help explain what's going on.

**Review by whytewolf — DISMISSED** [review]

**Review by waynew — CHANGES_REQUESTED** [review]
Okay, I opened a PR against your fork and I've also added I think what would make the docstring :+1: - if you want to finish this up I can close mine in favor of this PR, or vice versa and I can go ahead and combine these changes with my PR.

Those smol changes and this is ready to rock! :tada:

File: doc/topics/hardening.rst [hunk 2]
```diff
@@ -150,3 +150,13 @@ Run the following on the Salt minion:
 
 .. _salt-users: https://groups.google.com/forum/#!forum/salt-users
 .. _salt-announce: https://groups.google.com/forum/#!forum/salt-announce
+
+
+Hardening of syndic setups
+==========================
+
+Syndics must be run as the same user as their syndic master process. The master
+of master's will include publisher ACL information in jobs send to downstream
```
Comment (Ch3LL) [inline doc/topics/hardening.rst:159]:
```suggestion
of master's will include publisher ACL information in jobs sent to downstream
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Comment by Ch3LL** [comment]
Looks like the test failures are possibly related to this change.

**Review by whytewolf — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**